File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/SteamItem.java
package com.example.steaminvestmentmanager.utilclasses;

import android.graphics.Bitmap;
import com.google.gson.Gson;

public class SteamItem {
    private static final Gson gson = new Gson();
    private final String market_hash_name;
    private final String itemIconURL;
    private final String appid;
    private Bitmap itemIcon;
    private float currentPrice;
    private float buyingPrice;
    private int amount;
    private int currentCurrency;
    private String currentCurrencyMedianPrice;

    public SteamItem(String market_hash_name, String appid, String itemIconURL, String enteredPrice, String enteredAmount) {
        this.market_hash_name = market_hash_name;
        this.appid = appid;
        this.itemIconURL = itemIconURL;
        this.itemIcon = new DownloadBitmapImage(this.getItemIconURL(), "/215fx215").call();
        this.buyingPrice = Float.parseFloat(enteredPrice);
        this.amount = Integer.parseInt(enteredAmount);
        currentCurrency = CurrencyData.getCurrency();
    }

    public void updateItemPrice() {
        String priceURL = PriceURLCreator.getURL(this, true);
        String jsonSteamItem = new DownloadingPageHtmlCode(priceURL).call();
        PriceoverviewSteamItem priceoverviewSteamItem = gson.fromJson(jsonSteamItem, PriceoverviewSteamItem.class);
        if (priceoverviewSteamItem.isSuccess()) {
            this.setCurrentPrice(CurrencyData.transformPriceToNumber(priceoverviewSteamItem.getCurrentPrice()));
            this.setCurrentCurrencyLowestPrice(priceoverviewSteamItem.getCurrentPrice());
        }
    }

    public void updateIcon() {
        itemIcon = new DownloadBitmapImage(this.getItemIconURL(), "/215fx215").call();
    }

    public void setBuyingPrice(Float buyingPrice) {
        this.buyingPrice = buyingPrice;
    }

    public String getMarket_hash_name() {
        return market_hash_name;
    }

    public Float getCurrentPrice() {
        return currentPrice;
    }

    public Bitmap getItemIcon() {
        return itemIcon;
    }

    public void setItemIcon(Bitmap itemIcon) {
        this.itemIcon = itemIcon;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }

    public String getItemIconURL() {
        return itemIconURL;
    }

    public String getAppID() {
        return appid;
    }

    public int getAmount() {
        return amount;
    }

    public void setCurrentCurrency(int currentCurrency) {
        this.currentCurrency = currentCurrency;
    }

    public void setCurrentPrice(Float currentPrice) {
        this.currentPrice = currentPrice;
    }

    public int getCurrentCurrency() {
        return currentCurrency;
    }

    public void setCurrentCurrencyLowestPrice(String currentCurrencyMedianPrice) {
        this.currentCurrencyMedianPrice = "";
        this.currentCurrencyMedianPrice += currentCurrencyMedianPrice;
    }

    public String getCurrentCurrencyLowestPrice() {
        return currentCurrencyMedianPrice;
    }

    public Float getBuyingPrice() {
        return buyingPrice;
    }

    public boolean checkForBeingFull() {
        if (market_hash_name != null && itemIconURL != null && appid != null) {
            return !market_hash_name.equals("") && !itemIconURL.equals("") && !appid.equals("") && (amount >= 0) && (currentCurrency >= 0);
        }
        return false;
    }

    @Override
    public String toString() {
        return "m" + market_hash_name + "\n" +
                itemIconURL + "\n" +
                appid + "\n" +
                buyingPrice + "\n" +
                amount + "\n" +
                currentCurrency + "\n" +
                currentCurrencyMedianPrice;
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/SteamItem.kt
package com.example.steaminvestmentmanager.utilclasses

import android.graphics.Bitmap
import android.util.Log
import com.google.gson.Gson

class SteamItem(val market_hash_name: String?, val appID: String?, val itemIconURL: String?, enteredPrice: String, enteredAmount: String) {
    var itemIcon: Bitmap?
    var currentPrice = 0f
    var buyingPrice: Float
    var amount: Int
    var currentCurrency: Int

    init {
        itemIcon = DownloadBitmapImage(itemIconURL, "/215fx215").call()
        buyingPrice = enteredPrice.toFloat()
        amount = enteredAmount.toInt()
        currentCurrency = CurrencyData.getCurrency()
        updateItemPrice()
    }

    fun updateItemPrice() {
        val priceURL = PriceURLCreator.getURL(this, true)
        val jsonSteamItem = DownloadingPageHtmlCode(priceURL).call()
        val poSteamItem = gson.fromJson(jsonSteamItem, PriceoverviewSteamItem::class.java) ?: return
        if (poSteamItem.success) {
            currentPrice = CurrencyData.transformPriceToNumber(poSteamItem.lowest_price)
        }else {
            Log.d("SteamIvan", "Couldn't update Item Price $priceURL $jsonSteamItem $this")
        }
    }

    fun updateIcon() {
        itemIcon = DownloadBitmapImage(itemIconURL, "/215fx215").call()
    }

    fun checkForBeingFull(): Boolean {
        return if (market_hash_name != null && itemIconURL != null && appID != null) {
            market_hash_name != "" && itemIconURL != "" && appID != "" && amount >= 0 && currentCurrency >= 0
        } else false
    }

    override fun toString(): String {
        return """
            m$market_hash_name
            $itemIconURL
            $appID
            $buyingPrice
            $amount
            $currentCurrency
            $currentPrice
            """.trimIndent()
    }

    companion object {
        private val gson = Gson()
    }
}